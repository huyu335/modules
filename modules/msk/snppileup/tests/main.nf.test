nextflow_process {

    name "Test SNPPILEUP"
    script "../main.nf"
    process "SNPPILEUP"
    config "./nextflow.config"

    tag "modules"
    tag "modules_mskcc"
    tag "snppileup"

    test("snppileup - bam") {

        when {
            process {
                """                
                meta = [ id:'test' ] // meta map
                input[0] = [
                                meta,
                                file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam'], checkIfExists: true),
                                file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam_bai'], checkIfExists: true)                               
                           ]
                input[1] = [
                                meta,
                                file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_markduplicates_sorted_bam'], checkIfExists: true),
                                file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_markduplicates_sorted_bam_bai'], checkIfExists: true)
                                
                           ]
                input[2] = [
                                meta,
                                file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz'], checkIfExists: true),
                                file(params.test_data['homo_sapiens']['genome']['dbsnp_146_hg38_vcf_gz_tbi'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
