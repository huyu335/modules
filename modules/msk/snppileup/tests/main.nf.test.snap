{
    "snppileup - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-01-29T15:06:34.676228"
    },
    "snppileup - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,deac57ec8a27969028af701840d2b0ee"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,deac57ec8a27969028af701840d2b0ee"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-01-29T15:06:28.546836"
    }
}